File: 2022/1/adv_2022_1_2.cpp
#include <iostream>
#include <fstream>
#include <queue>
#include <vector>

int main() 
{
    auto input = std::ifstream("input1.txt");

    std::priority_queue<int> p;
    int cal = 0;
    for( std::string line; getline( input, line ); )
    {
        if (line == ""){
            p.push(cal);
            cal = 0;
        }else{
            cal = cal + stoi(line);
        }
    }
    int res = p.top();
    p.pop();res += p.top();p.pop();res += p.top();

    std::cout << res << std::endl;
    return 0;
}

File: aoc2022/1/adv_2022_1_2.py
import os
from queue import PriorityQueue

q = PriorityQueue()
cal = 0
with open(os.path.join(os.path.dirname(os.path.abspath(__file__)), 'input1.txt')) as fp:
    for line in fp:
        row = line.strip()
        if row == "":
            q.put(-cal)
            cal = 0
        else:
            cal = int(row) + cal
        
print(-(q.get() + q.get() + q.get()))
